### PR TITLE
GI-157: Profileinstellungen

### PR DESCRIPTION
**Pull request by noadjamila**
## Description

This PR implements the user's settings page. It allows users to change their password and delete their profile. The route is accessible only to authenticated users.
Some layout improvements and refactoring were also made.

## Jira Ticket

Link to the Jira ticket: [GI-157](https://techstudyfinder.atlassian.net/browse/GI-157)

## Author Checklist

- [x] Code follows project coding standards and conventions
- [x] Tests have been written or updated
- [x] All tests pass locally
- [x] No sensitive data or secrets are included
- [x] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

### CODE REVIEW

**Review by DamarisLi — CHANGES_REQUESTED** [review]
Funktioniert super! Das Einzige was mir aufgefallen ist, ist dass im Backend die service layer geskippt wurde. Bei change password ist die business logic im controller, die sollte in einen service. Bei delete user wäre das einfach nur ein Aufruf für consistency.
Beim Login flow ist es auch so, das ist mir erst jetzt aufgefallen aber das kann man sich später nochmal anschauen 🥲

File: client/src/components/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * @param props.children - The component to render if user is authenticated
+ * @returns The protected component or redirect to home
+ */
+export default function ProtectedRoute({ children }: { children: ReactNode }) {
+  const { user, isLoading } = useAuth();
+
+  // Wait for auth state to load
+  if (isLoading) {
+    return null; // or a loading spinner if you have one
```
[the remark this answers is not in the document]
Comment (DamarisLi) [reply client/src/components/ProtectedRoute.tsx]: Ich habe bei den Ergebnissen "Lädt..." wenn sie noch laden!

**Comment by DamarisLi** [comment]
Also wenn es kein großer Aufwand ist gerne, aber kann sonst auch nochmal separat in ein cleanup Ticket oder so.
Copilot hat hier einiges kommentiert haha, test coverage wäre ganz gut aber beim Rest könntest du einfach schauen was du für sinnvoll hältst, das waren hauptsächlich Kleinigkeiten

**Comment by noadjamila** [comment]
Habe alles umgesetzt! Für das login, logout, etc. habe ich schon angefangen zu refactorn, aber dann hatte ich keine Lust mehr :D 
Ich habe mir ja schon die US zum Design Fix gezogen, vielleicht kann ich das da einfach noch dazu nehmen! @DamarisLi

**Review by DamarisLi — APPROVED** [review]
Klasse! Yes das passt dann gut in die andere Story :)
